File: MajorityElement.cs
Given an array of size n, find the majority element. The majority element is the element that appears more than ⌊ n/2 ⌋ times.

You may assume that the array is non-empty and the majority element always exist in the array.

public class Solution {
    public int MajorityElement(int[] nums) {
            var val = 0;
            var count = 0;
            foreach (var t in nums)
            {
                if (count==0)
                {
                    val = t;
                    count = 1;
                }
                else if (t==val)
                {
                    count++;
                }
                else
                {
                    count--;
                }
            }
            return val;
    }
}

File: MajorityElement.go
package main

import "fmt"

func majorityElement(nums []int) int {
	fst := nums[0]
	cnt := 0
	for _, v := range nums {
		if v == fst {
			cnt++
		} else {
			cnt--
			if cnt == 0 {
				fst = v
				cnt = 1
			}
		}
	}
	return fst
}

func main() {
	fmt.Print(majorityElement([]int{1, 2, 1, 1, 1, 2, 2}))
}
